### ISSUE TITLE
Spotify embed not working

### ISSUE THREAD

**Issue by ariel-maidana** [Automattic/wp-calypso#68494]
### Quick summary

Embedding Spotify content (song or podcast episode) doesn't work. This issue affects the Spotify block, the Embed block, and the Audio block.

### Steps to reproduce

1. Create a new page.
2. Add a Spotify block.
3. Paste a podcast episode URL.
4. Click on embed.

### What you expected to happen

The episode should be embedded.

### What actually happened

A message saying that the content can't be embedded appears, along with two buttons to "try again" and "convert to link".

### Context

5597227-zd-woothemes

### Platform (Simple, Atomic, or both?)

Simple, Atomic

### Reproducibility

Consistent

### Available workarounds?

Partial workaround to embed Spotify episodes in Atomic sites only: Add a Custom HTML block and paste the Spotify embed code.

**Comment by ariel-maidana** [comment]
Please follow up with 5597227-zen once the issue is solved.

**Comment by ariel-maidana** [comment]
Another report here: 5595833-zd-woothemes
Also affects Anchor embeds.
Please follow up with the user once it's solved.

**Comment by wiesenhauss** [comment]
New reports here: 
5597612-zen
5597624-zen

**Comment by eduardozulian** [comment]
Another report: 5597648-zen

Another workaround is to use the [Shortcode block](https://wordpress.com/support/wordpress-editor/blocks/shortcode-block/). Example: `[spotify https://open.spotify.com/episode/2WMa1esNtw69RFNplCwdY9?si=64d2df849d37417f]`

(cc @ariel-maidana in case you'd like to update your report!)

**Comment by freelylovinglife** [comment]
Another report here: 5597049 -zd Please follow up with user once resolved.

**Comment by pauljacobson** [comment]
I have a report in 5596579-zen

I created a test page here: p9Nurx-Bo-p2

I could add an embed for a podcast episode in the editor, although only a link appeared on the post itself. I didn't have a problem embedding a music track though. An embed for the podcast episode using the link for episode failed.

https://user-images.githubusercontent.com/2912202/193445578-bdd9020d-3bfb-46c3-8ef2-dfe1f90c09ae.mp4

On the test post, this is the embed code for the embed that rendered as a link:

```
<figure class="wp-block-embed is-type-rich is-provider-spotify wp-block-embed-spotify wp-embed-aspect-21-9 wp-has-aspect-ratio"><div class="wp-block-embed__wrapper">
<a href="https://href.li/?https://open.spotify.com/episode/6Rj5MTRcUjmg96JsTfbWWp?si=28db7da3bcf945f2" rel="nofollow noreferrer">https://open.spotify.com/episode/6Rj5MTRcUjmg96JsTfbWWp?si=28db7da3bcf945f2</a>
</div></figure>
```

By contrast, this is the embed code for the music track that did embed:

```
<figure class="wp-block-embed is-type-rich is-provider-spotify wp-block-embed-spotify wp-embed-aspect-21-9 wp-has-aspect-ratio"><div class="wp-block-embed__wrapper">
<div class="embed-spotify"><iframe title="Spotify Embed: Obi-Wan" style="border-radius: 12px" width="100%" height="152" frameborder="0" allowfullscreen="" allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture" loading="lazy" src="https://open.spotify.com/embed/track/22VtvEbttc3y0MuekCm7Sb?si=4320f433b0d442f7&amp;utm_source=oembed"></iframe></div>
</div></figure>
```

**Comment by essleeung** [comment]
Another case here: 5600183-zd-woothemes

Shared the shortcode as a workaround.

- [ ] Update user once fix is done.

**Comment by wpshellbelle** [comment]
Another report at 5600488-zen 

Provided shortcode workaround.

**Comment by aetta** [comment]
+1  zd-5598017

Shared the shortcode workaround too!

**Comment by NoHopeRadio** [comment]
+1 30892895-hc

Gave shortcode instructions! User does want to know when this is updated.

**Comment by NoHopeRadio** [comment]
+1 5600539-zen

**Comment by sharonlaker19** [comment]
- [ ] 5602168-zen

**Comment by Sandkorner** [comment]
5602202-zen

**Comment by chrisanez** [comment]
5596097-zen

**Comment by cometgrrl** [comment]
@Automattic/apex Could y'all take a look at this?

**Comment by rinazrina** [comment]
5604457-zd-woothemes

**Comment by margaretschneider** [comment]
We've also gotten a report of this on a Newspack site. I tried the shortcode workaround and that didn't work for me, but an iframe embed worked. 

One nuance I'm not seeing noted: The embed via URL is previewable on the back end in the editor and on front-end preview for draft posts on this Newspack site, but once the post is published, only the link text of the Spotify URL appears on the front end. 

And for me, front-end preview of the shortcode added to a draft post in a shortcode block on a Newspack site only shows the shortcode text, not the embed.

**Comment by ivan-ottinger** [comment]
🤔 I wasn't able to reproduce the issue in my tests. I tried with two different Simple sites and one Atomic one.

These are the Spotify links I tried to embed:
- `https://open.spotify.com/episode/4fW6hdI3CAFsZR8nvExW8Z?si=bf77f7c8cc6f4ffa`
- `https://open.spotify.com/episode/6Rj5MTRcUjmg96JsTfbWWp?si=28db7da3bcf945f2`

The player embedded correctly and displays in the editor and published post as well.

Can anyone still reproduce the issue?

**Comment by margaretschneider** [comment]
I see what you're seeing, @ivan-ottinger, that these embeds are working again. I just tried this Spotify embed as well, from the original article where I'd reported it: 

- `https://open.spotify.com/episode/5oYMGuza5FIxB1KxvQCKoX`

This one and the ones you shared are all working in a published test article on the Newspack Model Home.

**Comment by ivan-ottinger** [comment]
Thank you for checking as well, @margaretschneider. 🙂 

In this case I am closing the issue, but we can reopen it in case we get additional reports.

**Comment by sajmes** [comment]
Very similar issue here. Except no error. Embedding Spotify playlists or songs simply turns the embed into a link.

I tried several songs and playlists, and tried on two Simple sites.

User report: 5776371-zd-woothemes

Shortcode workaround works for now.

**Comment by happychait** [comment]
Re-opening this issue as the normal Spotify embed is still not working.

Another issue in 37161114-zen

Workarounds:

1. Using the Shortcode block
2. Using the "Embed track" code on a Custom HTML block.

![Spotify Embed Track](https://user-images.githubusercontent.com/1519466/206616118-1aaab8cb-42c9-4fd8-964b-911bcd967312.jpeg)

**Comment by liviopv** [comment]
Another report in:

- [ ] 5776154-zen

It looks like the embed is broken for all kinds of Spotify media, not only episodes or tracks. Here's a link to my test: https://liviofree.wordpress.com/spotify-test/

Not a core issue, as it works fine on my local site:

![CleanShot 2022-12-09 at 11 06 54@2x](https://user-images.githubusercontent.com/14153300/206689050-e5ad71e4-8a54-43c7-82c1-824809f2c2c1.png)

**Comment by kimerlin81** [comment]
And another one - 34341161-hc
shared shortcode workaround; tested = works fine

Spotify Link: https://open.spotify.com/track/4mWR1rGkh5FEG7WOzespZP?si=1f41eb6394e34fffa

**Comment by upwardmomentum84** [comment]
Reported here: 37608914-hc

**Comment by zdenys** [comment]
5775375-zd-woothemes Recommended the Shortcode workaround like this:

 `[spotify https://open.spotify.com/track/50Cf2eYv8zT3v2HAkwhIiL?si=31bf469a643042fa]`

**Comment by syhussaini** [comment]
5780252-zen
Recommended a workaround of the shortcode indicated by @zdenys [here](https://github.com/Automattic/wp-calypso/issues/68494#issuecomment-1345250305).

**Comment by TeniCola** [comment]
Another case in 5776813-zen -- I recommended the shortcode solution to the user.

FWIW, I've had past moments with my personal sites where the Spotify block has worked on and off for some playlists. It was not consistent and was hard to reproduce at the time, but it was inconvenient when the embeds stopped displaying out of nowhere.

Eventually, I used the same solution to replace all my embedded playlists with a shortcode block, and I haven't had the problem since -- so hopefully the workaround can serve as a long-term solution, even while the Spotify block is under troubleshooting.

**Comment by zachspears** [comment]
Hello @ariel-maidana, It looks like the last report of this issue was 2 Weeks ago (Dec 11th) after a steady influx of reports. I am also unable to reproduce the issue (See below). It looks like this issue has been resolved but I wanted to know if you are still seeing it? Thanks!

![Screenshot on 2022-12-22 at 20-48-44](https://user-images.githubusercontent.com/10124024/209267523-ac3dc521-8d7a-41bf-8d27-75bcf05c68d2.jpg)

**Comment by kavyagokul** [comment]
📌 SCRUBBING : RESULT
- Tested on P2 - Could Not Replicate 
- Tested on Simple - Could Not Replicate 
- Tested on AT - Could Not Replicate 
- Tested on Self-hosted - Could Not Replicate  
- Replicable on Core - No

📌 ACTIONS
- Closed as could not replicate (please reopen the issue reoccurs)

📌 Message to Author
- CLosing this as multiple reports of not being able to repro anymore.

**Comment by ariel-maidana** [comment]
@zachspears I didn't encounter this issue lately. Just did a test on a test site and worked as expected. Thank you!
